### PR TITLE
Use name from server when displaying create/update

### PR DESCRIPTION
**Pull request by smarterclayton**
Allows generated names from files to be displayed so users can
interact with them.

### CODE REVIEW

File: pkg/kubectl/resource/visitor.go [hunk 1]
```diff
@@ -97,6 +98,37 @@ func (i *Info) Get() error {
 	return nil
 }
 
+// Refresh updates the object with another object. If ignoreError is set
+// the object will be updated directly
```
Comment (jlowdermilk) [inline pkg/kubectl/resource/visitor.go]: I think this should say "If ignoreError is set, the object will be updated even if metadata cannot be read from the object". The object will still be updated directly if ignoreError is False and metadata can be read.

File: pkg/kubectl/cmd/update.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ Examples:
 
   $ cat pod.json | kubectl update -f -
   <update a pod based on the json passed into stdin>
-  
+
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/update.go]: This whitespace cleanup is what's causing travis/shippable to complain. Run `hack/run-gendocs.sh`

**Comment by jlowdermilk** [comment]
LGTM, minor nits.

**Comment by smarterclayton** [comment]
Fixed

**Pull request merged**